app/src/main/java/com/example/newsportal/data/NewsRepository.kt

```diff
@@ -10,19 +10,20 @@ class NewsRepository(private val mService: NewsService, private val newsDao: New
     val data = newsDao.findAll()
 
     suspend fun refresh() {
-        for (elem in Categories.values()){
-            val news = mService.getTopNewsByCategory(elem.name).articles.map {
+        for (elem in Categories.values()) {
+            val news = mService.getTopNewsByCategory(elem.name).articles?.map {
                 ArticleLocal(
-                    category= elem.name,
+                    category = elem.name,

```

Comment:
suggest to left fields of ArcticleLocal non-nullable, it will be easy to work with that values later. 
Here you could map data from nullable to non-nullable with default values (e.g. String.orEmpty() for strings
)